lib/web/desktop.go

```diff
@@ -120,23 +113,8 @@ func (h *Handler) createDesktopConnection(
 	if username == "" {
 		return sendTDPError(trace.BadParameter("missing username"))
 	}
-	width, err := strconv.Atoi(q.Get("width"))
```

Comment:
Yes, instead of passing it by url param to the proxy, and then [having the proxy send it as a TDP message](https://github.com/gravitational/teleport/blob/ad734be859c84afb378173ab947bf005a897b729/lib/web/desktop.go#L208-L211), we're instead just connecting to the proxy via websocket and then sending it as a TDP message [straight from the browser client](https://github.com/gravitational/teleport/blob/bdb9b6b196df070e2dd929f7ef25c3af913f4373/web/packages/teleport/src/lib/tdp/client.ts#L117-L119) (same as all the other TDP messages) as soon as the connection is made.

Reply:
Does our backend do okay if we see it twice? Just thinking about compatibility and upgrades where you may see one from the browser client and one from a proxy.

Reply:
Good point which I hadn't considered. I ran a test where I send the message twice and everything works fine -- the only time we actually do anything with a tdp `ClientScreenSpec` message is in `readClientSize`, which picks up the first time we send this message:

https://github.com/gravitational/teleport/blob/185a064d81dd72b8b74097e3d2da487b89a1ae9c/lib/srv/desktop/rdp/rdpclient/client.go#L218

Any subsequent messages are handled by the regular user input streaming worker goroutine, which just ignores `ClientScreenSpec`s. You end up seeing them in this log line:

https://github.com/gravitational/teleport/blob/185a064d81dd72b8b74097e3d2da487b89a1ae9c/lib/srv/desktop/rdp/rdpclient/client.go#L628